test-tsd/querybuilder.test-d.ts

```diff
@@ -23,5 +23,7 @@ knex.QueryBuilder.extend('customSelect', function (value: number) {
 });
 
 const main = async () => {
-  expectType<number[]>(await knexInstance('users').customSelect<any, number[]>(42));
+  expectType<number[]>(
+    await knexInstance('users').customSelect<any, number[]>(42)
```

Comment:
You have error here:

Argument of type 

<TRecord extends {} = any, TResult = unknown[]>(this: QueryBuilder<TRecord, TResult>, **value: number**) => QueryBuilder<TRecord, TResult | QueryBuilder<TRecord, TResult>>

 is not assignable to parameter of type 

<TRecord extends {} = any, TResult = unknown[]>(this: QueryBuilder<TRecord, TResult>, **...args: any[]**) => QueryBuilder<TRecord, TResult> | Promise<QueryBuilder<TRecord, TResult>>.